### ISSUE TITLE
Perpetual diff with data sources (aws_kms_key and aws_iam_policy_document)

### ISSUE THREAD

**Issue by juliosantos** [hashicorp/terraform-provider-aws#16239]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

```
Terraform v0.13.5
+ provider registry.terraform.io/hashicorp/archive v2.0.0
+ provider registry.terraform.io/hashicorp/aws v3.15.0
+ provider registry.terraform.io/hashicorp/external v2.0.0
+ provider registry.terraform.io/hashicorp/local v2.0.0
+ provider registry.terraform.io/hashicorp/null v3.0.0
+ provider registry.terraform.io/hashicorp/random v3.0.0
+ provider registry.terraform.io/hashicorp/template v2.2.0
+ provider registry.terraform.io/jianyuan/sentry v0.6.0
```

### Affected Resource(s)
* aws_kms_key
* aws_iam_policy_document

### Terraform Configuration Files

```
variable "kms_master_key_id" {}

data "aws_kms_key" "master" {
  key_id = var.kms_master_key_id
}

data "aws_iam_policy_document" "send" {
  statement {
    actions = [
      "sqs:SendMessage",
    ]

    resources = [
      aws_sqs_queue.this.arn,
    ]
  }

  statement {
    actions = [
      "kms:DescribeKey",
      "kms:GenerateDataKey",
      "kms:Decrypt",
    ]

    resources = [
      data.aws_kms_key.master.arn,
    ]
  }
}
```

### Expected Behavior

I would expect to see these data resources in my plan never or once.

### Actual Behavior

Data resources are read and shown in the plan perpetually:

```
  # module.id.module.octopus.module.wallet_check_queue.data.aws_iam_policy_document.send will be read during apply
  # (config refers to values not yet known)
 <= data "aws_iam_policy_document" "send"  {
      + id      = "2275136200"
      + json    = jsonencode(
            {
              + Statement = [
                  + {
                      + Action   = "sqs:SendMessage"
                      + Effect   = "Allow"
                      + Resource = "arn:aws:sqs:eu-west-1:997625618900:frctls-staging-id-octopus-wallet-check-queue"
                      + Sid      = ""
                    },
                  + {
                      + Action   = [
                          + "kms:GenerateDataKey",
                          + "kms:DescribeKey",
                          + "kms:Decrypt",
                        ]
                      + Effect   = "Allow"
                      + Resource = "arn:aws:kms:eu-west-1:997625618900:key/0a6fc529-391c-4bdd-b7ae-086a10013bc1"
                      + Sid      = ""
                    },
                ]
              + Version   = "2012-10-17"
            }
        )
      + version = "2012-10-17"

      + statement {
          + actions       = [
              + "sqs:SendMessage",
            ]
          + effect        = "Allow"
          + not_actions   = []
          + not_resources = []
          + resources     = [
              + "arn:aws:sqs:eu-west-1:997625618900:frctls-staging-id-octopus-wallet-check-queue",
            ]
        }
      + statement {
          + actions       = [
              + "kms:Decrypt",
              + "kms:DescribeKey",
              + "kms:GenerateDataKey",
            ]
          + effect        = "Allow"
          + not_actions   = []
          + not_resources = []
          + resources     = [
              + "arn:aws:kms:eu-west-1:997625618900:key/0a6fc529-391c-4bdd-b7ae-086a10013bc1",
            ]
        }
    }

  # module.id.module.octopus.module.wallet_check_queue.data.aws_kms_key.master will be read during apply
  # (config refers to values not yet known)
 <= data "aws_kms_key" "master"  {
      + arn                      = "arn:aws:kms:eu-west-1:997625618900:key/0a6fc529-391c-4bdd-b7ae-086a10013bc1"
      + aws_account_id           = "997625618900"
      + creation_date            = "2018-12-08T16:51:05Z"
      + customer_master_key_spec = "SYMMETRIC_DEFAULT"
      + description              = "frctls-staging-id-octopus-key"
      + enabled                  = true
      + id                       = "0a6fc529-391c-4bdd-b7ae-086a10013bc1"
      + key_id                   = "0a6fc529-391c-4bdd-b7ae-086a10013bc1"
      + key_manager              = "CUSTOMER"
      + key_state                = "Enabled"
      + key_usage                = "ENCRYPT_DECRYPT"
      + origin                   = "AWS_KMS"
    }

Plan: 0 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

module.id.module.octopus.module.wallet_check_queue.data.aws_kms_key.master: Reading...
module.id.module.octopus.module.wallet_check_queue.data.aws_kms_key.master: Read complete after 0s [id=0a6fc529-391c-4bdd-b7ae-086a10013bc1]
module.id.module.octopus.module.wallet_check_queue.data.aws_iam_policy_document.send: Reading...
module.id.module.octopus.module.wallet_check_queue.data.aws_iam_policy_document.send: Read complete after 0s [id=2275136200]

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.
Releasing state lock. This may take a few moments...

Outputs:

ray_admin_password = <sensitive>
```

### Steps to Reproduce

1. `terraform apply`

### References

* https://github.com/hashicorp/terraform-provider-aws/issues/14579

**Comment by radimsuckr** [comment]
Hi, I'm having the exact same issue as @juliosantos with `aws_iam_policy_document` below.

Output of `terraform --version`:
```
Terraform v0.13.5
+ provider registry.terraform.io/hashicorp/archive v2.0.0
+ provider registry.terraform.io/hashicorp/aws v3.21.0
+ provider registry.terraform.io/hashicorp/external v2.0.0
+ provider registry.terraform.io/hashicorp/local v2.0.0
+ provider registry.terraform.io/hashicorp/null v3.0.0
+ provider registry.terraform.io/hashicorp/random v3.0.0
+ provider registry.terraform.io/hashicorp/template v2.2.0
+ provider registry.terraform.io/hashicorp/tls v3.0.0
```

```terraform
data "aws_iam_policy_document" "lambda" {
  statement {
    sid    = "AllowWritingToCloudWatchLogs"
    effect = "Allow"

    actions = [
      "logs:CreateLogGroup",
      "logs:CreateLogStream",
      "logs:PutLogEvents",
    ]

    resources = [
      "arn:aws:logs:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:log-group:/aws/lambda/${aws_lambda_function.refresh.function_name}:*",
    ]
  }

  statement {
    sid       = "AllowSSMGetParameter"
    effect    = "Allow"
    actions   = ["ssm:GetParameter"]
    resources = ["arn:aws:ssm:${data.aws_region.current.name}:*:parameter${var.ami_ssm_parameter}"]
  }
}
```

**Comment by radimsuckr** [comment]
Luckily upgrade to 0.14 required really small effort so I tried `0.14.2` and can confim that this issue no longer occurs on this version.

```
Terraform v0.14.2
+ provider registry.terraform.io/hashicorp/archive v2.0.0
+ provider registry.terraform.io/hashicorp/aws v3.21.0
+ provider registry.terraform.io/hashicorp/external v2.0.0
+ provider registry.terraform.io/hashicorp/local v2.0.0
+ provider registry.terraform.io/hashicorp/null v3.0.0
+ provider registry.terraform.io/hashicorp/random v3.0.0
+ provider registry.terraform.io/hashicorp/template v2.2.0
+ provider registry.terraform.io/hashicorp/tls v3.0.0
```

**Comment by breathingdust** [comment]
This looks like it was resolved in #14579 so closing. Please comment to reopen if this is not the case! 🚀
